src/renderer.cpp

```diff
@@ -469,26 +467,26 @@ void Renderer::recalcInternals()
         }
     }
 
-    if ((viewWidth % game_width) > 0 || (viewHeight % game_height) > 0)
+    // If internal_resolution_scale from settings is less than one, calculate the closest fit for the output resolution, otherwise use the value directly
+    if (internal_resolution_scale < 1)

```

Comment:
After some testing with @satsukiyatoshi he found out that the autoscaling works only once, when you run the game, then doesn't matter how many times you resize the window, this value will never change.

This is because the IF here, despite the function code being called everytime the window resolution changes, it will never be hit the moment it gets assigned a value inside the if.

Because of this you need to find another way to make sure this code is being hit everytime IF the user decided to put 0 as value. Also, please find a way that doesn't bring another flag in the toml file.

Thank you!